code/game/objects/items/breakable.dm

```diff
@@ -0,0 +1,260 @@
+	health-=thisdmg
+	play_hit_sounds(thisdmg)
+	if(!thisdmg)
+		return 0 //return 0 if the item took no damage (glancing attack)
+	else
+		if(health>0) //Only if the item isn't ready to break.
+			message_take_hit(mute)
+		damaged_updates()
+		return 1 //return 1 if the item took damage
+
+/obj/item/proc/play_hit_sounds(var/thisdmg, var/hear_glanced = TRUE, var/hear_damaged = TRUE) //Plays any relevant sounds whenever the item is hit. glanced_sound overrides damaged_sound if the latter is not set or hear_damaged is set to FALSE.
+	if(health<=0) //Don't play a sound here if the item is ready to break, because sounds are also played by on_broken().
+		return
+	if(thisdmg && !isnull(damaged_sound) && hear_damaged)
+		playsound(src, damaged_sound, 50, 1)
+	else if(!isnull(glanced_sound) && hear_glanced)
+		playsound(src, glanced_sound, 50, 1)
+
+/obj/item/proc/message_take_hit(var/mute = FALSE) //Give a visible message when the item takes damage.
+	if(!isnull(take_hit_text2) && !mute)
+		visible_message("<span class='warning'>\The [src] [pick(take_hit_text2)]!</span>")
+
+/obj/item/proc/damaged_updates() //Put any damage-related changes to name, desc, icon, etc. here.
+	return
+
+/obj/item/examine(mob/user, var/size = "", var/show_name = TRUE, var/show_icon = TRUE)
+	..()
+	if(health<maxHealth && damaged_examine_text)
+		user.simple_message("<span class='info'> [damaged_examine_text]</span>",\
+			"<span class='notice'> It seems kinda messed up somehow.</span>")
+
+/obj/item/proc/transfer_item_blood_data(obj/item/A,obj/item/B)	//Transfers blood data from one item to another.
+	if(!A || !B)
+		return
+	if(A.had_blood)
+		B.blood_color = A.blood_color
+		B.blood_DNA = A.blood_DNA
+		B.had_blood = TRUE
+
+		if(!blood_overlays[B.type]) //If there isn't a precreated blood overlay make one
+			B.generate_blood_overlay()
+
+		if(B.blood_overlay != null) // Just if(blood_overlay) doesn't work.  Have to use isnull here.
+			B.overlays.Remove(B.blood_overlay)
+		else
+			B.blood_overlay = blood_overlays[B.type]
+
+		B.blood_overlay.color = B.blood_color
+		B.overlays += B.blood_overlay
+
+/obj/item/proc/generate_break_text(var/glanced = FALSE, var/suppress_glance_text) //Generates text for when an item is hit.
+	if(glanced)
+		if(suppress_glance_text)
+			return "!"
+		else if(glances_text)
+			return ", but it [pick(glances_text)] off!"
+		else
+			return ", but it [pick("bounces","gleams","glances")] off!"
+	else if(health > 0 && !isnull(take_hit_text))
+		return ", [pick(take_hit_text)] it!"
+	else
+		return "!" //Don't say "cracking it" if it breaks because on_broken() will subsequently say "The item shatters!"
+
+/obj/item/proc/break_item(var/atom/target, var/range, var/speed, var/override , var/fly_speed, var/hit_atom) //Breaks the item if its health is 0 or below. Passes throw-related parameters to on_broken() to allow for an object's fragments to be thrown upon breaking.
```

Comment:
This proc is rather poorly-named. It checks if the item should break and breaks it if so, while the name would suggest that it just breaks it.